app/javascript/components/pages/Jobs.jsx

```diff
@@ -11,15 +12,20 @@ import 'stylesheets/jobs';
 
 const Jobs = () => {
     const [jobs, setJobs] = useState([]);
+    const [cookies] = useCookies('token');
 
     useEffect(() => {
         const fetchData = async () => {
-            const data = await getJobs();
-            setJobs(data);
+            try {
+                const data = await getJobs(cookies['token']);
+                setJobs(data);
+            } catch (error) {
+                window.location.href = '/jobs/authenticate';
+            }
```

Comment:
This is the new logic for fetching jobs -- it passes in the JWT stored in cookies, and if an error is raised (i.e. there's an issue with the token), it redirects to the `jobs/authenticate` page.